### ISSUE TITLE
spotcheck warnings from kokkos 

### ISSUE THREAD

**Issue by kyungjoo-kim** [kokkos/kokkos-kernels#284]
Warnings from kokkos

```
/home/kyukim/Work/lib/kokkos/master/core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp(645): 
warning: variable "mask" was declared but never referenced    
/home/kyukim/Work/lib/kokkoskernels/build/spotcheck/TestAll_2018-09-05_15.24.03/cuda/9.0.103/Cuda_OpenMP-release/kokkos/install/include/Cuda/Kokkos_Cuda_ReduceScan.hpp(871): 
warning: dynamic initialization in unreachable code 
```
This is probably related to @crtrott 's new reduction.

**Comment by crtrott** [comment]
Yeah it is. I tried to fix that but so far without much luck (at least as a solution which fixes it for all compiler versions). I need to figure out the right ifdef combinations ....

**Comment by ndellingwood** [comment]
Looks like Dan fixed the second-half with kokkos/kokkos#1820.

**Comment by ndellingwood** [comment]
This is now fixed in Kokkos develop branch.
